sudo/lib/exec/parent.rs

```diff
@@ -212,15 +216,41 @@ impl ParentClosure {
             }
         }
     }
+
+    /// Handle changes to the monitor status.
+    fn handle_sigchld(&mut self) {
+        const OPTS: WaitOptions = WaitOptions::new().all().untraced().no_hang();
+
+        let (pid, status) = loop {
+            match waitpid(WaitPid::any(), OPTS) {
+                Err(WaitError::Io(err)) if was_interrupted(&err) => {}
+                Err(_) => return,
```

Comment:
I guess this branch reachable because one can send fake SIGCHLD using `kill`? "fake" as in not associated to a child exiting / stopping. perhaps that would be worthwhile to mention as a comment

Reply:
Yeah that should only happen if we receive SIGCHLD and no child process has a status update to report, it could be someone sending SIGCHLD to us or if some "race" happens. I'm not sure how common the latter is but I'd consider it should be very rare.